searx/settings.yml

```diff
@@ -121,6 +121,7 @@ ui:
 #
 # result_proxy:
 #   url: http://127.0.0.1:3000/
+#   # use !!binary to write the key in base64 otherwise use an usual string
```

Comment:
Hint: did you see comment started in line 117 .. in cfcbc3a5c3 you added a note about `!!binary` notation where you point to commit af77ec3 .. but this commit does not exists.

I would prefer ..

    # the key is a base64 encoded string, the YAML !!binary prefix is optional

and line 119 needs to be fixed: *(keep the !!binary notation)* .. is no longer valid.

Do we need a documentation of this setting? / morty is no longer needed as image proxy and we keep the option to configure `proxy` link silently.

Reply:
It has two different meaning.
Both are `foobar`:
* `!!binary "Zm9vYmFy"`
* `"foobar"` <-- here the key is not base64 encoded

( !!binary encode the string with base64 : https://yaml.org/type/binary.html )

Reply:
> So `!!binary` is not optional, up to the user to choose between raw string or base64 encoded string.

Oops sorry, but if so, how can it help to solve the issue #1522 ...

> When generating the settings.yml from NixOS code ``!!binary`` cannot easily be set.

If I understand @SuperSandro2000 correctly there has been a base64 string generated `"Zm9vYmFy"` this base64 string is used when the morty service is started (`morty -key Zm9vYmFy ...`)

The morty key is always a base64 string / see https://github.com/asciimoo/morty#usage

And he wants to use this string notation in settings.yml, whithout the need to add a `!!binary` prefix

```yaml
result_proxy:
  url: http://127.0.0.1:3000/
  key: Zm9vYmFy
  proxify_results: true
```

.. or I am wrong?

Reply:
> The morty key is always a base64 string

yes, now I remember that is has created issues few year ago. For example:
* user were writing the key as plain text in morty, which was decoded as base64.
* then copy / paste the key in settings.yml  and it didn't worked.

As far I understand Nix can't generated the `!!binary` prefix, so this PR should solve that : just skip the prefix, it works the same.

Reply:
> I have updated the PR in this way.

OK, LGTM .. there is a question leftover from my initial comment 

[Do we need a documentation of this setting? / morty is no longer needed as image proxy and we keep the option to configure `proxy` link silently.](https://github.com/searxng/searxng/pull/1708#discussion_r956693217)

What do you think?

Reply:
About pathlib, I'm going to skip this change for now.

> Do we need a documentation of this setting?

No need for documentation, settings.yml is good enough here: 
* it doesn't break existing installation
* this PR makes the life easier with Nix, nothing more

Reply:
For me a feature is official when it is documented .. if there is no documentation, only a comment in a config file it is an hidden feature (without version stability).

Since `result_proxy` is based on a no longer maintained software (morty) its fine for me to keep it unofficial.